File: elixir/easy/3005/3005.exs
defmodule Solution do
  @spec max_frequency_elements(nums :: [integer]) :: integer
  def max_frequency_elements(nums) do
    frequencies =
      nums
      |> Enum.frequencies()

    max_value =
      frequencies
      |> Map.values()
      |> Enum.max()

    frequencies
    |> Enum.filter(fn {_k, v} -> v == max_value end)
    |> Enum.map(fn {_k, v} -> v end)
    |> Enum.sum()
  end
end

ExUnit.start()

defmodule AssertionTest do
  use ExUnit.Case, async: true

  test "test max_frequency_elements" do
    assert Solution.max_frequency_elements([1, 2, 2, 3, 1, 4]) == 4
    assert Solution.max_frequency_elements([1, 2, 3, 4, 5]) == 5
  end
end

File: py/easy/3005/3005.py
from collections import Counter
from typing import List


class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        frequencies = Counter(nums)
        max_frequency = max(frequencies.values())
        filtered = filter(lambda x: x[1] == max_frequency, frequencies.items())
        mapped = map(lambda x: x[1], filtered)
        return sum(mapped)


solution = Solution()
assert solution.maxFrequencyElements([1, 2, 2, 3, 1, 4]) == 4
assert solution.maxFrequencyElements([1, 2, 3, 4, 5]) == 5
